azext_iot/common/fileops.py

```diff
@@ -0,0 +1,47 @@
+from os import makedirs, remove, listdir
+from os.path import exists, join
+from pathlib import PurePath
+from typing import Union
+from azure.cli.core.azclierror import FileOperationError
+
+
+"""
+fileops: Functions for working with files.
+"""
+
+
+def write_content_to_file(
+    content: Union[str, bytes],
+    destination: str,
+    file_name: str,
+    overwrite: bool = False,
+):
+    dest_path = PurePath(destination)
+    file_path = dest_path.joinpath(file_name)
+
+    if exists(file_path) and not overwrite:
+        raise FileOperationError(f"File already exists at path: {file_path}")
+    if overwrite and destination:
+        makedirs(destination, exist_ok=True)
+    write_content = bytes(content, "utf-8") if isinstance(content, str) else content
+    with open(file_path, "wb") as f:
+        f.write(write_content)
+
+
+def tar_directory(
```

Comment:
why do we need tar files...

Reply:
The end result of `edge devices create` if an `--output` is provided is both the Hub configuration changes (create device, set parents, set modules, etc) as well as a device bundle to install on edge devices themselves, to setup local edge agent configs and certificate paths.

Reply:
So since most edge use-cases are on linux endpoints/devices - we decided to use tar to compress the device bundles.